sql_metadb/derived_tables/po_lines_location.sql

```diff
@@ -17,25 +18,51 @@ WITH ploc AS (
 )
 SELECT
     ploc.pol_id,
-    ploc.pol_loc_id,
     ploc.pol_loc_qty,
     ploc.pol_loc_qty_elec,
     ploc.pol_loc_qty_phys,
-    loc.name AS location_name
+CASE WHEN ploc.pol_location_id IS NOT NULL THEN ploc.pol_location_id  
+          ELSE hr.permanent_location_id
+    END AS pol_location_id, 
+    CASE WHEN loc.name IS NOT NULL THEN loc.name
+         ELSE loc2.name
+    END AS pol_location_name,
+    CASE WHEN loc.name IS NOT NULL THEN 'pol_location'
+         WHEN loc2.name IS NOT NULL THEN 'pol_holding'
+         ELSE 'no_source'
+    END AS pol_location_source
 FROM
     ploc
-    LEFT JOIN folio_inventory.location__t AS loc ON loc.id = ploc.pol_loc_id;
+    LEFT JOIN folio_inventory.holdings_record__t AS hr ON ploc.pol_holding_id::uuid= hr.id
+    LEFT JOIN folio_inventory.location__t AS loc ON loc.id = ploc.pol_location_id
+    LEFT JOIN folio_inventory.location__t AS loc2 ON loc2.id ::uuid= hr.permanent_location_id ;
 
 CREATE INDEX ON po_lines_locations (pol_id);
 
-CREATE INDEX ON po_lines_locations (pol_loc_id);
-
 CREATE INDEX ON po_lines_locations (pol_loc_qty);
 
 CREATE INDEX ON po_lines_locations (pol_loc_qty_elec);
 
 CREATE INDEX ON po_lines_locations (pol_loc_qty_phys);
 
-CREATE INDEX ON po_lines_locations (location_name);
+CREATE INDEX ON po_lines_locations (pol_location_id);
+
+CREATE INDEX ON po_lines_locations (pol_location_name);
+
+CREATE INDEX ON po_lines_locations (pol_location_source);
+
+COMMENT ON COLUMN po_lines_locations.pol_id IS 'UUID identifying this purchase order line';
+
+COMMENT ON COLUMN po_lines_locations.pol_loc_qty IS 'combined/total quanitity of physical and electronic items';
+
+COMMENT ON COLUMN po_lines_locations.pol_loc_qty_elec IS 'Quantity of electronic items in this purchase order line';
+
+COMMENT ON COLUMN po_lines_locations.pol_loc_qty_phys IS 'Quantity of physical items or resources of "Other" order format in this purchase order line';
+
+COMMENT ON COLUMN po_lines_locations.pol_location_id IS 'UUID of the (inventory) location record or Holding UUID associated with order line';
+
+COMMENT ON COLUMN po_lines_locations.pol_location_name IS 'Name of the location assosiated with pol_location_id';
```

Comment:
`assosiated` should be `associated`

Reply:
Could you clarify what still needs to be address on this PR?

Reply:
I suggested many changes in my initial review. 

https://github.com/folio-org/folio-analytics/pull/697#pullrequestreview-1201801632

My main concern is updating the query to help explain/document the logic changes better. I also identified line by line several typos and style inconsistencies that could be updated at the same time.